### ISSUE TITLE
Some navigation xsheet/timeline requests (as after effects)

### ISSUE THREAD

**Issue by openanim** [opentoonz/opentoonz#4560]
### Is your feature request related to a problem? Please describe

I would like to see some features and shortcuts available in after effects  in opentoonz to navigate quickly and efficiently in the xsheet/timeline.

### Describe the solution you'd like

1) An editable timecode usable with separator(s) and + and - keys similar to after effects

In after effects we can navigate on timeline by typing directly in the timecode [ 0:00:00:00 ] 
Serverals writing mode are available. We can use several separators, either the . or , or ; with the numpad to navigate, also using + and - to move forward or backward a certain number of frames or time and it is very efficient. Something similar in OT would make it easy to navigate and make it more accurate especially when you have little visual space.

Exemple : navigate with a separator that represents the timecode example the .

1 => 0:00:00:01
1. => 0:00:01:00
1.. => 0:1:00:00
1... => 1:00:0:00
1.5 => 0:00:1:05
etc.
---------------------------------------
be able to add or decrease time with + and -

1. => 0:00:01:00
+5 => 0:00:01:05
+1. => 0:00:02:05
+-5 =>0:00:02:00
+-1. => 0:00:01:00
etc.

------------------------------------------

2) reach first frame level and last frame level

Navigation at the beginning and end of a level. As well as reaching the beginning of the next or previous level in the same column. useful when the levels do not start and stop at the same time.

![1 first frame level last frame level](https://user-images.githubusercontent.com/29790512/195964229-2592fefa-399d-4d86-801a-f958051ffb10.gif)

3) Be able to select columns using the numpad (0-999)

Typing a number selects the column corresponding to the number. Typing multiple digits in a row for columns requiring multiple digits. 0 to select the camera column and 1-999 for all others.

![6 choose columns with numpad](https://user-images.githubusercontent.com/29790512/195964282-d455d8a1-3deb-41bb-ae03-759676a892a5.gif)

4) Snap the first frame or the last frame of the level to the time indicator 

If there is enough space, move the start or end of a level to the time indicator

![2 start and end of layer at the curent time indicator](https://user-images.githubusercontent.com/29790512/195964321-cdb21c45-79db-4bb1-be8a-3c73e743c506.gif)

5) Adjust the working area to the size of the level

Shortcut to adjust the playback area to the size of the selected level.

![5 adjust work area to the level](https://user-images.githubusercontent.com/29790512/195964459-a3c9ca20-ff48-482a-b445-7095854f75d7.gif)

6) split the layer at the time indicator

![3 split level at time indicator](https://user-images.githubusercontent.com/29790512/195964415-0b33196a-a2c5-4c4b-b9d7-f265827fead4.gif)

7) Pop up timecode to reach a specific point in the xsheet/timeline

Especially helpful when working in full screen.

**Comment by flurick** [comment]
For point nr 2, apparently shift+up/down keys does something like this. Which is a bit confusing for the horizontal mode.

**Comment by openanim** [comment]
Yes, shift arrow up/down has some similarities but is not really reliable because it takes empty cells into account and does not take the whole level into account. Very often we are not at the right place either on the level or on the column. The idea is really to reach the first and the last frame of the level even if it has other drawings in it. Although in my example I only stretched a single drawing, but the behavior should be similar if the level has several drawings.  

In terms of "behavior" it's a bit like using the "next drawing" and "previous drawing" shortcuts on a level containing a drawing at the beginning and end only (as the gif below). But here it is as if these commands only work on the first and last frame of any level, and we can only reach that.

![reach first and last frame level](https://user-images.githubusercontent.com/29790512/200192880-c24bd240-fa2b-4d44-9000-7cee8ba7c674.gif)

Ideally, this should be a new commands to avoid altering any current workflow.

**Comment by flurick** [comment]
Took at look at how the current commands are implemented. One thing I found is that there is A last frame command in the "customize xhsheet toolbar" menu called "[Last Frame](https://github.com/opentoonz/opentoonz/blob/1166e8ab8f55329274dacf3ec9c2007989892f36/toonz/sources/toonz/mainwindow.cpp#L2007)" (for example), which in turn calls the viewer to [go to its last frame](https://github.com/opentoonz/opentoonz/blob/3a536dd46946efb278cf8768171762d894c3d099/toonz/sources/toonzqt/flipconsole.cpp#L1552) (so it only works with the viewer open)

**Comment by openanim** [comment]
@flurick 

Thank you for your interest in this request. I use the first frame and last frame commands in OT, the ones in the play menu (in shortcuts). I guess maybe it's because of the playback controls that the viewer is needed. I also use start, end and page commands, in the same way, more versatile. Unfortunately they don't really help in point n 2. 

Although I'm not sure I understand what you mean by individual windows. In my opinion the current commands should remain unchanged for now, at least in relation to this request to avoid altering any existing workflow. 

In after effects for point 2 refers to the commands called "Go to layer In point (i) and " Go to layer Out point (o) as well as Go to previous In point or Out point (Ctrl+Alt+Shift+Left Arrow), and Go to next In point or Out point (Ctrl+Alt+Shift+Right Arrow) to get to the next points. I don't know if this helps in describing how this request could works.

**Comment by RodneyBaker** [comment]
For now... transferring this report to Opentoonz-Docs as the original reporter has ghosted themselves.
